### DISCUSSION TITLE
Support for pyodbc.Cursor.execute with 0 or more parameters

### DISCUSSION THREAD

**Discussion by CapAdv** [mkleehammer/pyodbc#1235] [Q&A] [answered]
### Environment

- Python: 3.11.4, 64-bit
- pyodbc: 4.0.35
- OS: Windows 
- DB: MS SQL Server 2016 (build 13.0.5026)
- driver:  ODBC Driver 18 for SQL Server

### Issue

If I execute an sql command (its MS SQL Server 2016 as specified above) using pyodbc and pass 'None' as the parameters, I'm expecting it to run a command that takes no parameters, like "waitfor delay '00:00:10'" without error, but actually it gives:

pyodbc.ProgrammingError: ('The SQL contains 0 parameter markers, but 1 parameters were supplied', 'HY000'). 

```
sql = "waitfor delay '00:00:10'"
sql_params = None
with pyodbc.connect(settings.Application_Conn, autocommit=True) as conn:
    crsr = conn.cursor()
    crsr.execute(sql, sql_params)

```

Is there a way using pyodbc to include the sql_params parameter in the crsr.execute but allow it to be 'empty' somehow and not fail when called against a command which takes no parameters, or maybe this is already possible ?

**Comment by gordthompson** [comment, marked as answer]
```python
sql = "SELECT 1 AS foo"
crsr.execute(sql, None)
```

is equivalent to

```python
sql = "SELECT 1 AS foo"
crsr.execute(sql, (None,))
```

which are both ways to pass a single parameter value of `None` (which maps to SQL `NULL`). To pass no parameters, simply use an empty tuple

```
sql = "SELECT 1 AS foo"
crsr.execute(sql, tuple())
```

**Reply by CapAdv** [reply]
That'll do it. Thanks.
